state/modelgeneration.go

```diff
@@ -74,13 +77,99 @@ func (g *Generation) AssignedUnits() map[string][]string {
 // AssignApplication indicates that the application with the input name has had
```

Comment:
How do we know if units have been added to the generation before changes to the application have been made?  Related to a question in AssignUnit().

Reply:
Insightful. 

The modelling as is means that if a generation was created and all units of an application added to it with no configuration change, then it would be auto-completed.

On the other hand, if we flagged changed application config separately, the rules would mean that for the scenario above, auto-completion would happen upon the first config change, which seems like a stronger violation of reasonably expected behaviour.

I will land this patch, and we can discuss when you are online.